webrender/src/platform/unix/font.rs

```diff
@@ -98,66 +101,120 @@ impl FontContext {
-                              (!enable_aa && bitmap.pixel_mode as i8 == FT_PIXEL_MODE_MONO as i8));
+        None
+    }
 
-                let mut final_buffer = Vec::with_capacity((bitmap.width * bitmap.rows) as usize * 4);
+    #[allow(dead_code)]     // TODO(gw): Expose this to the public glyph dimensions API.
+    pub fn get_glyph_dimensions(&self,
+                                font_key: FontKey,
+                                size: Au,
+                                character: u32,
+                                device_pixel_ratio: f32) -> Option<(Au, Au)> {
+        self.load_glyph(font_key, size, character, device_pixel_ratio).map(|slot| {
+            let metrics = unsafe { &(*slot).metrics };
+            (Au::from_px((metrics.width >> 6) as i32), Au::from_px((metrics.height >> 6) as i32))
+        })
+    }
 
-                if enable_aa {
-                    let buffer = slice::from_raw_parts(
-                        bitmap.buffer,
-                        (bitmap.width * bitmap.rows) as usize
-                    );
+    pub fn get_glyph(&mut self,
+                     font_key: FontKey,
+                     size: Au,
+                     character: u32,
+                     device_pixel_ratio: f32,
+                     render_mode: FontRenderMode) -> Option<RasterizedGlyph> {
+        let mut glyph = None;
+
+        if let Some(slot) = self.load_glyph(font_key,
+                                            size,
+                                            character,
+                                            device_pixel_ratio) {
+            let render_mode = match render_mode {
+                FontRenderMode::Mono => FT_RENDER_MODE_MONO,
+                FontRenderMode::Alpha => FT_RENDER_MODE_NORMAL,
+                FontRenderMode::Subpixel => FT_RENDER_MODE_LCD,
+            };
 
-                    // Convert to RGBA.
-                    for &byte in buffer.iter() {
-                        final_buffer.extend_from_slice(&[ 0xff, 0xff, 0xff, byte ]);
-                    }
-                } else {
-                    // This is not exactly efficient... but it's only used by the
-                    // reftest pass when we have AA disabled on glyphs.
-                    for y in 0..bitmap.rows {
-                        for x in 0..bitmap.width {
-                            let byte_index = (y * bitmap.pitch) + (x >> 3);
-                            let bit_index = x & 7;
-                            let byte_ptr = bitmap.buffer.offset(byte_index as isize);
-                            let bit = (*byte_ptr & (128 >> bit_index)) != 0;
-                            let byte_value = if bit {
-                                0xff
-                            } else {
-                                0
-                            };
-                            final_buffer.extend_from_slice(&[ 0xff, 0xff, 0xff, byte_value ]);
+            unsafe {
```

Comment:
Good idea - how about I do that as a follow up PR today (I have some more changes to the font code coming).